libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/data/MapboxNavigationViewportDataSource.kt

```diff
@@ -382,6 +392,8 @@ class MapboxNavigationViewportDataSource(
             cameraForFollowing = followingCameraOptions,
             cameraForOverview = overviewCameraOptions
         )
+
+        Log.d("MapZoomingTest", "MapboxNavigationViewportDataSource.evaluate(). options: $options, resulting viewportData: $viewportData")
```

Comment:
Let's also log:
1. `pointsToFrameOnCurrentStep`, `targetLocation`, `additionalPointsToFrameForFollowing` and `pointsToFrameAfterCurrentStep` in `updateFollowingData`;
2. `simplifiedRemainingPointsOnRoute`, `targetLocation`, `additionalPointsToFrameForOverview` in `updateOverviewData`.

My camera intuition says the problem is somewhere in here, so it'd be nice to have as many logs as possible.